src/core/SampleBuffer.cpp

```diff
@@ -1017,30 +1017,61 @@ void SampleBuffer::visualize(
 	const float ySpace = h * 0.5f;
 	const int nbFrames = focusOnRange ? toFrame - fromFrame : m_frames;
 
-	const int fpp = qBound<int>(1, nbFrames / w, 20);
-	QPointF * l = new QPointF[nbFrames / fpp + 1];
-	QPointF * r = new QPointF[nbFrames / fpp + 1];
+	const int fpp = qMax<int>(1, nbFrames / w);
+	std::vector<QPointF> fMax(nbFrames / fpp * 2 + 2);
+	std::vector<QPointF> fRms(nbFrames / fpp * 2 + 2);
 	int n = 0;
 	const int xb = dr.x();
 	const int first = focusOnRange ? fromFrame : 0;
 	const int last = focusOnRange ? toFrame - 1 : m_frames - 1;
 
 	for (int frame = first; frame <= last; frame += fpp)
 	{
+		float maxData = -1;
+		float minData = 1;
+
+		float rmsData[2] = {0, 0};
+
+		// Find maximum and minimum samples within range
+		for (int i = 0; i < fpp && frame + i < last; ++i)
+		{
+			for (int j = 0; j < 2; ++j)
+			{
+				if (m_data[frame + i][j] > maxData)
+				{
+					maxData = m_data[frame + i][j];
+				}
+
+				if (m_data[frame + i][j] < minData)
+				{
+					minData = m_data[frame + i][j];
+				}
+
+				rmsData[j] += m_data[frame + i][j] * m_data[frame + i][j];
```

Comment:
Lots of repeated same data access, again compiler should be smart to reuse, but wouldn't hurt to help and then make it a little more readable.
```suggestion
				auto curData = m_data[frame + i][j];
				
				if (curData > maxData) { maxData = curData; }
				if (curData < minData) { minData = curData; }

				rmsData[j] += curData * curData;
```